bin/cylc-print

```diff
@@ -89,8 +90,8 @@ def get_option_parser():
 
 @cli_function(get_option_parser)
 def main(parser, options, regfilter=None):
-    suite_src_files_mgr = SuiteSrvFilesManager()
-    allsuites = suite_src_files_mgr.list_suites(regfilter)
+    allsuites = list(get_scan_items_from_fs(
+        reg_pattern=regfilter, compile_reg=True, active_only=False))
```

Comment:
Could we `import re` in this file, then use `reg_pattern=re.compile(regfilter)`? This way we don't need the `compile_reg` flag.

Reply:
I thought about doing that, but that would mean the ``re.compile()`` is not wrapped in a ``try/except`` statement, as it was in the original function, so it seems safest to leave it as-is to catch any ``re.error``.

I also considered putting the ``try/except`` in ``cylc print`` rather than in this function, but decided against it for some reason I can't recall or see, but actually that would also avoid the flag *and* allow us to keep the ``try/except``, so I will use that approach instead as that seems to be best on both fronts!